flight/Modules/Battery/battery.c

```diff
@@ -149,14 +152,21 @@ static void batteryTask(void * parameters)
 					scaled_voltage = 0.0f;
 					adc_offset_invalid = true;
 				}
+				else if(scaled_voltage > 3.0f && !cells_calculated && (batterySettings.MaxCellVoltage > 0)){
+					cells_calculated = true;
+					cells = ((scaled_voltage / batterySettings.MaxCellVoltage) + 1);
+					flightBatteryData.DetectedCellCount = cells;
```

Comment:
I think we should use rounding instead of `((scaled_voltage / batterySettings.MaxCellVoltage) + 1)`. If for example `MaxCellVoltage` is 4.2 and the voltage sensor calibration is slightly off, the reading for a full 3S battery may be 12.65V, the current code would then detect 4 cells.